### PR TITLE
[Feature][File] Support config null format for text file read

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

Support config null format for text file read

### Does this PR introduce _any_ user-facing change?

add config key

### How was this patch tested?

added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by davidzollo — DISMISSED** [review]
+1
LGTM

File: docs/en/connector-v2/source/SftpFile.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ The File does not have a specific type list, and we can indicate which SeaTunnel
 | compress_codec            | String  | No       | None                | The compress codec of files and the details that supported as the following shown: <br/> - txt: `lzo` `None` <br/> - json: `lzo` `None` <br/> - csv: `lzo` `None` <br/> - orc: `lzo` `snappy` `lz4` `zlib` `None` <br/> - parquet: `lzo` `snappy` `lz4` `gzip` `brotli` `zstd` `None` <br/> Tips: excel type does Not support any compression format                            |
 | archive_compress_codec    | string  | no       | none                |
 | encoding                  | string  | no       | UTF-8               |
+| null_format               | string  | no       | -                   | Only used when file_format_type is text. null_format to define which strings can be represented as null.                                                                                                                                                                                                                                                                        |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/SftpFile.md]: provide an example? E.g. `\N`

**Comment by wuchunfu** [comment]
@hailin0 I have a question, shouldn't this specify a certain field? Is it specifying a certain data now?

File: docs/en/connector-v2/source/SftpFile.md (hunk 1 above)
Comment (hailin0) [reply docs/en/connector-v2/source/SftpFile.md]: https://stackoverflow.com/questions/43556160/null-data-converted-into-n-for-numeric-columns-in-hive

https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=27362034#LanguageManualDDL-StorageFormatsStorageFormatsRowFormat,StorageFormat,andSerDe

https://github.com/alibaba/DataX/blob/master/hdfsreader/doc/hdfsreader.md

**Comment by hailin0** [comment]
> @hailin0 I have a question, shouldn't this specify a certain field? Is it specifying a certain data now?

This is a common setting and has nothing to do with columns

File: docs/en/connector-v2/source/SftpFile.md (hunk 1 above)
Comment (Hisoka-X) [reply docs/en/connector-v2/source/SftpFile.md]: I mean put an example into the doc. Provide some common null format as example.

**Comment by wuchunfu** [comment]
> > @hailin0 I have a question, shouldn't this specify a certain field? Is it specifying a certain data now?
> 
> This is a common setting and has nothing to do with columns

Got it

File: docs/en/connector-v2/source/SftpFile.md (hunk 1 above)
Comment (hailin0) [reply docs/en/connector-v2/source/SftpFile.md]: fixed

**Review by Hisoka-X — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]

**Pull request merged**